src/main.rs

```diff
@@ -373,6 +375,55 @@ fn reply_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
     0
 }
 
+/// show subcommand implementation
+///
+fn show_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
+    // NOTE: we use boxed iterators since we might do a lot of dynamic wrapping
+    //       depending on command line parameters
+
+    // first, get us an iterator over all the commits
+    // NOTE: "issue" is a required parameter
+    let issue = try_or_1!(Oid::from_str(matches.value_of("issue").unwrap()));
+    let mut commits : Vec<(TreeGraphElemLine, Commit)> =
+        try_or_1!(repo.get_issue_revwalk(issue)).abort_on_err()
+            .map(|oid| repo.find_commit(oid)).abort_on_err()
+            .into_tree_graph()
+            .collect();
+
+    // Decide on the order in which the messages will be printed.
+    if matches.is_present("tree") {
+        // We want the commits in chronological order
+        commits.reverse();
+        for commit in commits.iter_mut() {
+            commit.0.reverse_marks();
+        }
+    };
```

Comment:
I'm not entirely sure whether this is a nice idea. We're introducing a _lot_ of overhead with this reverse thing here (because we first `collect` and then build a new iterator from it).

Using the reverse functionality from `itertools`, this would be _way_ more performant (no `collect`) and LLVM would be able to optimize this _way_ more agressively.

But, on the other hand, we have the problem that introducing a new iterator here for doing this is also a lot of code.

### Suggestion

I would like to fork your PR and optimize this, if you ACK me on this.